Sources/SmokeAWSHttp/AWSClientInvocationTraceContext.swift

```diff
@@ -62,6 +62,7 @@ public struct AWSClientInvocationTraceContext: InvocationTraceContext {
         if let response = response, response.status.code >= 500 && response.status.code < 600 {
             logger.error("\(logLine)")
         } else {
+            logger.error("\(error)")
```

Comment:
We should log the error in both cases (500 responses and other responses) at the appropriate level for the type of response. We don't want to log at error for a say a 400 response as it could pollute metrics around error log emissions with false positives.

The easiest way to do this would be to pass the error through to getLogLine and add it to the line that is being constructed in that function.